ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js

```diff
@@ -25,6 +25,30 @@ const store = require('../../js/store.js')
 const QuerySettingsView = require('../query-settings/query-settings.view.js')
 const user = require('../singletons/user-instance.js')
 
+import query from '../../react-component/utils/query'
+
+const getCacheKey = (attr, obj) =>
+  JSON.stringify(attr) + JSON.stringify(obj.src)
+
+const fetchFacets = memoize(async (attr, { src: srcs }) => {
+  const json = await query({
```

Comment:
should you add error handling here?

Reply:
If the network request fails then facets will end up being undefined and an empty array will be returned. Am I misunderstanding something about the way fetch requests work?

Reply:
Leaving an error message when the fetch request fails would help with debugging when unexpected behavior arises